test/test_message_filters_cache.py

```diff
@@ -55,108 +57,113 @@ def __init__(self):
 
 
 class TestCache(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        rclpy.init()
+        cls.node = rclpy.create_node('my_node', namespace='/my_ns')
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.node.destroy_node()
+        rclpy.shutdown()
 
     def test_all_funcs(self):
-        sub = Subscriber("/empty", String)
+        sub = Subscriber(self.node, String, "/empty")
         cache = Cache(sub, 5)
 
         msg = AnonymMsg()
-        msg.header.stamp = rospy.Time(0)
+        msg.header.stamp = Time(seconds=0)
         cache.add(msg)
 
         msg = AnonymMsg()
-        msg.header.stamp = rospy.Time(1)
+        msg.header.stamp = Time(seconds=1)
         cache.add(msg)
 
         msg = AnonymMsg()
-        msg.header.stamp = rospy.Time(2)
+        msg.header.stamp = Time(seconds=2)
         cache.add(msg)
 
         msg = AnonymMsg()
-        msg.header.stamp = rospy.Time(3)
+        msg.header.stamp = Time(seconds=3)
         cache.add(msg)
 
         msg = AnonymMsg()
-        msg.header.stamp = rospy.Time(4)
+        msg.header.stamp = Time(seconds=4)
         cache.add(msg)
 
-        l = len(cache.getInterval(rospy.Time(2.5),
-                                  rospy.Time(3.5)))
-        self.assertEquals(l, 1, "invalid number of messages" +
+        l = len(cache.getInterval(Time(seconds=2.5), Time(seconds=3.5)))
```

Comment:
Can you keep the wrapping for a smaller diff and easier reading in this and the next 3 sections.